src/main/kotlin/com/noumenadigital/npl/cli/commands/registry/CommandExecutor.kt

```diff
@@ -1,17 +1,48 @@
 package com.noumenadigital.npl.cli.commands.registry
 
 import com.noumenadigital.npl.cli.ExitCode
-import com.noumenadigital.npl.cli.commands.CommandParameter
+import com.noumenadigital.npl.cli.commands.NamedParameter
 import com.noumenadigital.npl.cli.service.ColorWriter
+import io.modelcontextprotocol.kotlin.sdk.Tool
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.put
+import kotlinx.serialization.json.putJsonObject
 
 interface CommandExecutor {
     val commandName: String
     val description: String
 
-    val parameters: List<CommandParameter>
+    val parameters: List<NamedParameter>
         get() = emptyList()
 
+    val supportsMcp: Boolean
+        get() = true
+
     fun createInstance(params: List<String>): CommandExecutor = this
 
     fun execute(output: ColorWriter): ExitCode
+
+    fun toMcpToolInput(): Tool.Input =
+        Tool.Input(
+            properties =
+                buildJsonObject {
+                    parameters.filter { !it.isHidden }.forEach { parameter ->
+                        putJsonObject(parameter.name) {
+                            if (parameter.takesValue) {
+                                put("type", "string")
+                            } else {
+                                put("type", "boolean")
+                            }
+                            val description =
+                                if (parameter.name == "sourceDir" || parameter.name == "target") {
+                                    "${parameter.description} (should be an absolute path)"
```

Comment:
this is a bit hacky, but still mostly fine, I think. 

The problem is that tools that use MCP typically don't run the CLI in the "current project directory" but rather somewhere completely different. This is different from how the CLI is typically executed.